course/include/isometry.h

```diff
@@ -0,0 +1,97 @@
+
+
+// Standard libraries
+#include <cmath>
+#include <initializer_list>
+#include <iostream>
+#include <limits>
+#include <sstream>
+#include <string>
+
+namespace ekumen {
+namespace math {
+
+static constexpr int resolution{4};
+template <class T>
+typename std::enable_if<!std::numeric_limits<T>::is_integer, bool>::type almost_equal(T a, T b, int resolution) {
+    return std::fabs(a - b) <= std::numeric_limits<T>::epsilon() * std::fabs(a + b) * resolution
+           || std::fabs(a - b) < std::numeric_limits<T>::min();
+}
+
+struct Elements {
+    Elements() : x_(0), y_(0), z_(0) {};
+    Elements(const double &x, const double &y, const double &z) : x_(x), y_(y), z_(z) {}; 
+    double x_,y_,z_;
+};
+
+class Vector3 {
+    public:
+        Vector3() : v_(new Elements{}) {};
+        Vector3(const double &x, const double &y, const double &z) : v_(new Elements{x, y, z}) {};
+        ~Vector3() {
+            // deallocate
+            delete v_;  
+        };
+
+        // copy constructor
+        Vector3(const Vector3& other) : v_(new Elements{*(other.v_)}) {};
+
+        // move constructor
+        Vector3(Vector3&& other) : v_(std::__exchange(other.v_, nullptr)) {};
```

Comment:
Better do: 
```
Vector3(Vector3&& other) {
  v_ = other.v_; // transfer ownership
  other.v_ = nullptr; // don't leave a copy of the pointer behind
}
```
or use `std::exchange`, not `std::__exchange`, but that's available since c++14.